app/chains/itinerary_chain.py

```diff
@@ -84,12 +95,80 @@ def extract_chat_history_content(x):
     chat_history=extract_chat_history_content
 ) | prompt | llm
 
+session_memories = {}
+
+def _cleanup_expired_sessions():
+    """
+    Remove session histories that have not been accessed within TTL to avoid memory leaks.
+    No-op if TTL is non-positive.
+    """
+    try:
+        ttl = SESSION_MEMORY_TTL_SECONDS
+        now = time.time()
+        expired_session_ids = []
+        for s_id, entry in list(session_memories.items()):
+            if not isinstance(entry, dict) or "last_access" not in entry:
+                continue
```

Comment:
The check for 'last_access' key skips entries silently, which may hide data corruption. Log a warning if an entry is missing expected fields.
```
if not isinstance(entry, dict) or "last_access" not in entry:
    logging.warning(f"Malformed session entry for {s_id}: {entry}")
    continue
```